src/traces/bar/cross_trace_calc.js

```diff
@@ -36,7 +36,8 @@ function crossTraceCalc(gd, plotinfo) {
         var fullTrace = fullTraces[i];
         if(
             fullTrace.visible === true &&
-            Registry.traceIs(fullTrace, 'bar') &&
+            (Registry.traceIs(fullTrace, 'bar') ||
+            Registry.traceIs(fullTrace, 'waterfall')) &&
```

Comment:
We should a new trace module category that encompasses `bar`, `histogram` and `waterfall` traces

https://github.com/plotly/plotly.js/blob/02e51d8f5ba5601f53ee3dc839ba05fb12c7f8a2/src/traces/bar/index.js#L31

or better yet (if it works out), maybe the `waterfall` trace module could be declared with the `'bar'` category?

Reply:
Yes the `waterfall` could be declared as `'bar'` category. 
Done in 56786530ff31d865aefd30ad4ebab4cff7487f94.